language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -0,0 +1,493 @@
+import org.eclipse.lsp4j.RenameParams;
+import org.eclipse.lsp4j.SaveOptions;
+import org.eclipse.lsp4j.ServerCapabilities;
+import org.eclipse.lsp4j.SignatureHelp;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.eclipse.lsp4j.TextDocumentPositionParams;
+import org.eclipse.lsp4j.TextDocumentSyncKind;
+import org.eclipse.lsp4j.TextDocumentSyncOptions;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.eclipse.lsp4j.WorkspaceSymbolParams;
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.lsp4j.launch.LSPLauncher;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.lsp4j.services.LanguageServer;
+import org.eclipse.lsp4j.services.TextDocumentService;
+import org.eclipse.lsp4j.services.WorkspaceService;
+import org.sonarsource.sonarlint.core.StandaloneSonarLintEngineImpl;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.Issue;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneAnalysisConfiguration;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration.Builder;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneSonarLintEngine;
+
+public class SonarLintLanguageServer implements LanguageServer, WorkspaceService, TextDocumentService {
+
+  private static final String DISABLE_TELEMETRY = "disableTelemetry";
+  static final String TEST_FILE_PATTERN = "testFilePattern";
+  static final String ANALYZER_PROPERTIES = "analyzerProperties";
+  private static final String SONARLINT_CONFIGURATION_NAMESPACE = "sonarlint";
+  private static final String SONARLINT_SOURCE = SONARLINT_CONFIGURATION_NAMESPACE;
+  private static final String SONARLINT_OPEN_RULE_DESCRIPTION_COMMAND = "SonarLint.OpenRuleDesc";
+  private static final String DISABLE_PROPERTY_KEY = "sonarlint.telemetry.disabled";
+
+  private final LanguageClient client;
+  private final StandaloneSonarLintEngine engine;
+  private final Future<?> backgroundProcess;
+  private final LanguageClientLogOutput logOutput;
+  private final Map<URI, String> languageIdPerFileURI = new HashMap<>();
+  private final SonarLintTelemetry telemetry = new SonarLintTelemetry();
+
+  private Path workspaceDir;
+  private String testFilePattern;
+  private Map<String, String> analyzerProperties;
+  private int ruleServerPort;
+
+  public SonarLintLanguageServer(int port) throws IOException {
+    Socket socket = new Socket("localhost", port);
+
+    Launcher<LanguageClient> launcher = LSPLauncher.createServerLauncher(this,
+      socket.getInputStream(),
+      socket.getOutputStream(),
+      true, new PrintWriter(System.out));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of constructor "PrintWriter(OutputStream)" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1943)